src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/ResourceContainerRepository.kt

```diff
@@ -0,0 +1,154 @@
+package org.wycliffeassociates.otter.jvm.persistence.repositories
+
+import io.reactivex.Completable
+import io.reactivex.schedulers.Schedulers
+import org.jooq.DSLContext
+import org.wycliffeassociates.otter.common.collections.tree.Tree
+import org.wycliffeassociates.otter.common.collections.tree.TreeNode
+import org.wycliffeassociates.otter.common.data.model.Collection
+import org.wycliffeassociates.otter.common.data.model.Content
+import org.wycliffeassociates.otter.common.domain.mapper.mapToMetadata
+import org.wycliffeassociates.otter.common.persistence.repositories.IResourceContainerRepository
+import org.wycliffeassociates.otter.jvm.persistence.database.AppDatabase
+import org.wycliffeassociates.otter.jvm.persistence.entities.ResourceLinkEntity
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.CollectionMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.ContentMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.LanguageMapper
+import org.wycliffeassociates.otter.jvm.persistence.repositories.mapping.ResourceMetadataMapper
+import org.wycliffeassociates.resourcecontainer.ResourceContainer
+
+class ResourceContainerRepository(
+        private val database: AppDatabase
+) : IResourceContainerRepository {
+    private val collectionDao = database.getCollectionDao()
+    private val contentDao = database.getContentDao()
+    private val resourceMetadataDao = database.getResourceMetadataDao()
+    private val languageDao = database.getLanguageDao()
+    private val resourceLinkDao = database.getResourceLinkDao()
+
+    override fun importResourceContainer(rc: ResourceContainer, rcTree: Tree, languageSlug: String): Completable {
+        return Completable.fromAction {
+            database.transaction { dsl ->
+                val language = LanguageMapper().mapFromEntity(languageDao.fetchBySlug(languageSlug, dsl))
+                val metadata = rc.manifest.dublinCore.mapToMetadata(rc.dir, language)
+                val dublinCoreFk = resourceMetadataDao.insert(ResourceMetadataMapper().mapToEntity(metadata), dsl)
+
+                // TODO: Should this only happen if we are importing a "help" RC? What if the help is imported first,
+                // TODO: ... then another related bundle RC is imported?
+                val relatedDublinCoreIds: List<Int> =
+                        linkRelatedResourceContainers(dublinCoreFk, rc.manifest.dublinCore.relation, dsl)
+
+                // TODO: Probably shouldn't hardcode "help"
```

Comment:
Re todo: I think the RC standard forces this. There's nothing in the manifest's `relation` that tells which resource is primary. tN links to ulb, and ulb links to tN.

Reply:
Joe corrected me. The manifest's `type: 'help'` has a special meaning in the RC spec, indicating a secondary collection.